### DISCUSSION TITLE
Getting the terminal title on update

### DISCUSSION THREAD

**Discussion by austincondiff** [migueldeicaza/SwiftTerm#302] [General]
I need to get the terminal title and update it whenever it changes so that I can display it in my view. Is there currently any way to do this? All I see in the API documentation is a `setTitle` function.

To give you a little context, I am creating tabbed UI for my terminals in [CodeEdit](https://github.com/CodeEditApp/CodeEdit) here in [this](https://github.com/CodeEditApp/CodeEdit/pull/1261) PR. As you can see in the image below, I'd like the title that you can see in Terminal.app, to be the title of my tab. When this title updates, I'd like it to be reflected in the tab's title.

<img width="1728" alt="Screenshot 2023-05-29 at 11 14 11 AM" src="https://github.com/migueldeicaza/SwiftTerm/assets/806104/afc8ed5a-44a0-4e35-9001-5c6b533ce4c0">

**Comment by migueldeicaza** [comment]
Yes there is a delegate method called when the title changes

**Reply by austincondiff** [reply]
And what method might that be? Could you point me in the right direction?

**Reply by austincondiff** [reply]
@migueldeicaza We have an issue for this in our repo [here](https://github.com/CodeEditApp/CodeEdit/issues/1298) and are discussing it [here](https://discord.com/channels/951544472238444645/954084547694325861/1115744080572457010) in Discord. If you wouldn't mind giving us a push in the right direction, we'd really appreciate it!
